picoquic/frames.c

```diff
@@ -170,6 +170,54 @@ int picoquic_flow_control_check_stream_offset(picoquic_cnx_t* cnx, picoquic_stre
     return ret;
 }
 
+
+/* ********* 
+ *
+ *  Frames
+ *
+ * ******** */
+
+typedef int (*picoquic_decode_frame_t)(picoquic_cnx_t* cnx, const uint8_t** pbytes, const uint8_t* bytes_max, int restricted, uint64_t current_time);
+
+static int picoquic_varint_skip_check(picoquic_cnx_t* cnx, const uint8_t** pbytes, const uint8_t* bytes_max)
+{
+    if (*pbytes < bytes_max) {
+        size_t length = picoquic_varint_skip(*pbytes);
+        if (length <= bytes_max - *pbytes) {
+            *pbytes += length;
+            return 0;
+        }
+    }
+    return picoquic_connection_error(cnx, PICOQUIC_TRANSPORT_FRAME_FORMAT_ERROR);
+}
+
```

Comment:
Yes, this can get refactored into single-return.  For functions that can fail in different ways along the way, it may mean adding some extra code and moving the common-case flow from the "top-level" into separate "then" blocks.  But it can be done systematically.